app/assess/models/ui/applicants_response.py

```diff
@@ -244,6 +256,26 @@ def _ui_component_from_factory(item: dict, application_id: str):
         if presentation_type not in ("grouped_fields",):
             item["answer"] = float(item["answer"])
 
+    # TODO : Handle "monthYearField" field convertion in a better way if exists in "multiInputField"
+    if (
+        answer
+        and field_type == "monthYearField"
+        and presentation_type == "text"
+    ) or (
+        answer and field_type == "multiInputField" and isinstance(answer, list)
+    ):
+        if isinstance(answer, list):
+            try:
+                if answer[1][2] == "monthYearField":
+                    input_date = answer[1][1][0]
+                    item["answer"][1][1][0] = _convert_to_month_year(
+                        input_date
+                    )
+            except IndexError:
+                pass
+        else:
+            item["answer"] = _convert_to_month_year(answer)
```

Comment:
This is pretty difficult to read, would it be good to add some test coverage for this so we know what's happening here?

Reply:
This is a tricky scenario to implement. Especially when `monthYearField` field is embedded some where in nested fields like in `multiInputField`. Atm it is a hack when it is nested in `multiInputField`. Updated the unittests to include this scenario